mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergFilterFactory.java

```diff
@@ -127,7 +127,7 @@ private static Object leafToLiteral(PredicateLeaf leaf) {
       case FLOAT:
         return leaf.getLiteral();
       case DATE:
-        return daysFromTimestamp((Timestamp) leaf.getLiteral());
+        return daysFromDate((Date) leaf.getLiteral());
```

Comment:
I think we should check the type returned by `getLiteral` and handle that here. Then we won't need separate code for different versions.

Reply:
Good point, I'll make that change.